cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -242,6 +243,20 @@ public void mapReduceInfo(HttpRequest request, HttpResponder responder,
       } catch (IOException ioe) {
         LOG.warn("Failed to get run history from JobClient for runId: {}. Falling back to Metrics system.", run, ioe);
         mrJobInfo = mapReduceMetricsInfo.getMRJobInfo(run);
+      } catch (NotFoundException nfe) {
+        // Even if we ran the MapReduce program, there is no guarantee that the JobClient will be able to find it.
+        // For example, if the MapReduce program fails before it successfully submits the job.
+        LOG.warn("Failed to find run history from JobClient for runId: {}. Falling back to Metrics system.", run, nfe);
```

Comment:
should be info. When job history is not there - it'll always spit out.. Ops will freak out for WARNS. Maaybe even debug